python/tk_desktop/desktop_window.py

```diff
@@ -990,13 +990,16 @@ def clear_app_uis(self):
         self.project_overlay.hide()
 
         # clear the project specific menu
+        self.clear_project_menu()
+        self.__pipeline_configuration_separator = None
+
+    def clear_project_menu(self):
```

Comment:
The code will have to be a bit more complicated then that. If you have multiple pipeline configurations, you will have both the pipeline configurations with their checkboxes next to them and the actions. The way you did this on Reload and Restart you will wipe all the menu. 

Here's what it looks like on my computer:

![screen shot 2017-11-08 at 08 08 34](https://user-images.githubusercontent.com/8126447/32550705-17077f58-c45c-11e7-9808-cffac533cabb.png)

You need to removing everything up to the separator and then leave everything else in. The method can then be renamed to something more specific like `clear_engine_actions_from_project_menu`.